s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/AWSV2SourceClient.java

```
@@ -80,17 +83,49 @@ public AWSV2SourceClient(final S3SourceConfig s3SourceConfig, final Set<String>
         this.s3Client = s3Client;
         this.bucketName = s3SourceConfig.getAwsS3BucketName();
         this.failedObjectKeys = new HashSet<>(failedObjectKeys);
+
+        // TODO the code below should be configured in some sort of taks assignement method/process/call.
+        int maxTasks;
+        try {
+            final Object value = s3SourceConfig.originals().get("tasks.max");
+            if (value == null) {
+                LOGGER.info("Setting tasks.max to 1");
+                maxTasks = 1;
+            } else {
+                maxTasks = Integer.parseInt(value.toString());
+            }
+        } catch (NumberFormatException e) { // NOPMD catch null pointer
+            LOGGER.warn("Invalid tasks.max: {}", e.getMessage());
+            LOGGER.info("Setting tasks.max to 1");
+            maxTasks = 1;
+        }
+        this.maxTasks = maxTasks;
+        int taskId;
+        try {
+            final Object value = s3SourceConfig.originals().get("task.id");
+            if (value == null) {
+                LOGGER.info("Setting task.id to 0");
+                taskId = 0;
+            } else {
+                taskId = Integer.parseInt(value.toString()) % maxTasks;
+            }
+        } catch (NumberFormatException e) { // NOPMD catch null pointer
+            LOGGER.warn("Invalid task.id: {}", e.getMessage());
+            LOGGER.info("Setting task.id to 0");
+            taskId = 0;
```

We shouldn't default this to 0 in my opinion, this would be a config error of some kind and should fail to start the connector, it should throw an error and stop the connector otherwise we could end up with multiple connector instances processing the same information.

---

This is going to be pulled out and dealt with TaskAssignment PR as it requires a lot of Test updates & config change.

---

See #384